samples/image-metadata/src/components/AppView/AppView.js

```diff
@@ -0,0 +1,184 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import camelCase from 'lodash.camelcase';
+import { Heading, Typography, Paragraph } from '@contentful/forma-36-react-components';
+
+import { ConfigurationContent } from './ConfigurationContent';
+import { InstallationContent } from './InstallationContent';
+import { Divider } from '../Divider';
+import { styles } from './styles';
+
+import appLogo from './app-logo.svg';
+
+const APP_INSTALLATION_STATE = {
+  LOADING: 'LOADING',
+  INSTALLED: 'INSTALLED',
+  NOT_INSTALLED: 'NOT_INSTALLED'
+};
+
+function convertToAppInstallationState(appIsInstalled) {
+  return appIsInstalled ? APP_INSTALLATION_STATE.INSTALLED : APP_INSTALLATION_STATE.NOT_INSTALLED;
+}
+
+export class AppView extends Component {
+  static propTypes = {
+    sdk: PropTypes.object.isRequired
+  };
+
+  state = {
+    allContentTypesIds: [],
+    appInstallationState: APP_INSTALLATION_STATE.LOADING,
+    contentTypeId: camelCase('Image Wrapper'),
+    contentTypeName: 'Image Wrapper'
+  };
+
+  async componentDidMount() {
+    const { sdk } = this.props;
+    const { app } = sdk.platformAlpha;
+
+    const appIsInstalled = await app.isInstalled();
+
+    const allContentTypes = await sdk.space.getContentTypes();
+    const allContentTypesIds = allContentTypes.items.map(({ sys: { id } }) => id);
+
+    // Following eslint error is caused due to using async/await
+    // eslint-disable-next-line react/no-did-mount-set-state
+    this.setState({
+      appInstallationState: convertToAppInstallationState(appIsInstalled),
+      allContentTypesIds
+    });
+
+    if (!appIsInstalled) {
+      app.onConfigure(this.installApp);
+    }
+  }
+
+  installApp = async () => {
+    const { allContentTypesIds, contentTypeId, contentTypeName } = this.state;
+
+    if (!contentTypeName) {
+      this.props.sdk.notifier.error('Provide a name for the content type.');
+      return false;
+    }
+
+    const isContentTypeIdTaken = allContentTypesIds.includes(contentTypeId);
+    if (isContentTypeIdTaken) {
+      this.props.sdk.notifier.error(
+        `ID "${contentTypeId}" is taken. Try a different name for the content type`
```

Comment:
ACtually this reminds me that we could also allow users to customize the ID (API name). IDs are used in CDN queries, SDKs etc. and some people might want to use some ID that follows their content type ID convention (instead of generated one). I'd render two inputs, one for name and one for ID, with ID automatically generated (`camelCase(name)` is fine) until you touch the ID field.

Reply:
Done